bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1623,25 +1624,22 @@ OperationResult PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgS
   requestsHandler_.preExecute(request, std::nullopt, cid, span);
   auto preProcessResult = static_cast<OperationResult>(request.outExecutionStatus);
   resultLen = request.outActualReplySize;
-  if (preProcessResult != OperationResult::SUCCESS) {
-    LOG_ERROR(logger(),
-              "Pre-execution failed" << KVLOG(
-                  cid, clientId, reqOffsetInBatch, reqSeqNum, (uint32_t)preProcessResult, resultLen));
-    return preProcessResult;
-  }
-  if (request.outActualReplySize == 0) {
+
+  if (!(request.outExecutionStatus) && request.outActualReplySize == 0) {
```

Comment:
1. We need to print an error log in both cases: preProcessResult != OperationResult::SUCCESS and 
request.outActualReplySize == 0.
2. We cannot be sure that in case request.outActualReplySize is 0 the result is not SUCCESS.